lib/IDE/CodeCompletion.cpp

```diff
@@ -4018,61 +3928,86 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
     return !candidates.empty();
   }
 
-  static bool collectPossibleArgTypes(DeclContext &DC, CallExpr *CallE,
-                                      Expr *CCExpr,
-                                      SmallVectorImpl<FunctionParams> &Candidates,
-                                      unsigned &Position, bool &HasName) {
-    if (!collectionInputTypes(DC, CallE, Candidates))
-      return false;
-
-    if (auto *tuple = dyn_cast<TupleExpr>(CallE->getArg())) {
-      for (unsigned i = 0, n = tuple->getNumElements(); i != n; ++i) {
-        if (isa<CodeCompletionExpr>(tuple->getElement(i))) {
-          HasName = !tuple->getElementName(i).empty();
-          Position = i;
-          return true;
-        }
-      }
-
-      return getPositionInTupleExpr(DC, CCExpr, tuple, Position, HasName);
-    } else if (isa<ParenExpr>(CallE->getArg())) {
+  static bool getPositionInArgs(DeclContext &DC, Expr *Args, Expr *CCExpr,
+                                unsigned &Position, bool &HasName) {
+    if (isa<ParenExpr>(Args)) {
       HasName = false;
       Position = 0;
       return true;
     }
 
+    auto *tuple = dyn_cast<TupleExpr>(Args);
+    if (!tuple)
+      return false;
+
+    for (unsigned i = 0, n = tuple->getNumElements(); i != n; ++i) {
+      if (isa<CodeCompletionExpr>(tuple->getElement(i))) {
+        HasName = !tuple->getElementName(i).empty();
+        Position = i;
+        return true;
+      }
+    }
+    auto &SM = DC.getASTContext().SourceMgr;
+    for (unsigned i = 0, n = tuple->getNumElements(); i != n; ++i) {
```

Comment:
Why do we have both of these loops?  Are there any cases found by the first loop that we couldn't find with the second?

Reply:
Not sure. I just inlined `getPositionInTupleExpr()` here, and I wanted to keep this PR strictly NFC. I'll do further improvements in follow-ups.